src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2694,7 +2694,30 @@ impl Coordinator {
 
             // Determine if fast path plan will be used for this explainee
             let fast_path_plan = match explainee {
-                Explainee::Query => peek::create_fast_path_plan(&mut dataflow, GlobalId::Explain)?,
+                Explainee::Query => {
+                    let timestamp_context = self
```

Comment:
We recently refactored peek into multiple stages, see `sequence_peek_stage` in this file. I'm assuming that `EXPLAIN` should behave identically to peek, except stop executing at a certain point. For example `EXPLAIN QUERY` stops right before sending the dataflow to compute (I just made this up, I'm not familiar with the actual explain stages). I think that `EXPLAIN` should use the same peek stages as peeks, but selectively opt out of certain stages.

Reply:
In that case I still think it makes sense for `EXPLAIN`s with a `SELECT` `EXPLAINEE` to re-use the peek stages to generate the appropriate metadata for the select. Otherwise we should just copy and past the logic.

Reply:
I managed to get this working.

Overall, keeping the compilation pipeline for the `sequence_<do something>` and `sequence_explain_plan` for that same thing in sync is getting a bit unwieldy. We have some tracking issues for the main problem from our side #18089, and hopefully some of those can be addressed by #18496. Hopefully we can allocate some time for that in the next months.